test/e2e/tkn_helper_utility.go

```diff
@@ -0,0 +1,316 @@
+package e2e
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"gotest.tools/icmd"
+	knativetest "knative.dev/pkg/test"
+
+	"knative.dev/pkg/test/logging"
+
+	//"github.com/tektoncd/cli/test/helper"
+	"github.com/tektoncd/pipeline/pkg/names"
+	"golang.org/x/xerrors"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	// Mysteriously by k8s libs, or they fail to create `KubeClient`s from config. Apparently just importing it is enough. @_@ side effects @_@. https://github.com/kubernetes/client-go/issues/242
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+)
+
+var initMetrics sync.Once
+
+const (
+	interval = 1 * time.Second
+	timeout  = 10 * time.Minute
+)
+
+func Setup(t *testing.T) (*Clients, string) {
+	t.Helper()
+	namespace := names.SimpleNameGenerator.RestrictLengthWithRandomSuffix("testcli")
+
+	initializeLogsAndMetrics()
+	c := NewClients(knativetest.Flags.Kubeconfig, knativetest.Flags.Cluster, namespace)
+	CreateNamespace(namespace, c.KubeClient)
+	VerifyServiceAccountExistence(namespace, c.KubeClient)
+	return c, namespace
+}
+
+func Header(logf logging.FormatLogger, text string) {
+	left := "### "
+	right := " ###"
+	txt := left + text + right
+	bar := strings.Repeat("#", len(txt))
+	logf(bar)
+	logf(txt)
+	logf(bar)
+}
+
+func SetupTektonPipelines() (*Clients, string) {
+	namespace := "tekton-pipelines"
+	initializeLogsAndMetrics()
+	c := NewClients(knativetest.Flags.Kubeconfig, knativetest.Flags.Cluster, namespace)
+	InstallTektonPipelines(c.KubeClient, namespace)
+	VerifyServiceAccountExistence(namespace, c.KubeClient)
+	log.Println("CRD Installation successful")
+
+	return c, namespace
+}
+
+func CleanUpResources() {
+	msg := "kubectl delete --ignore-not-found=true {{.Element}}.tekton.dev --all"
+	ar := [...]string{"pipelineresources", "tasks", "pipelines", "taskruns", "pipelineruns"}
+	vars := make(map[string]interface{})
+	for _, element := range ar {
+		vars["Element"] = element
+		cmd := ProcessString(msg, vars)
+		cmdArgs := strings.Fields(cmd)
+
+		res := icmd.RunCmd(icmd.Command(cmdArgs[0], cmdArgs[1:]...))
+
+		if res.ExitCode != 0 {
+			log.Fatalf("Failed to execute %s", res.Stderr())
+		}
+	}
+	log.Println("Cleaned up existing resources.. successfully")
+}
+
+func InstallTektonPipelines(kubeClient *knativetest.KubeClient, namespace string) {
+	if os.Getenv("PIPELINE_VERSION") == "" {
+		log.Printf("Environment Varaiable PIPELINE_VERSION is not set")
+		os.Setenv("PIPELINE_VERSION", "0.5.2")
+	}
+
+	Header(log.Printf, fmt.Sprintf("Installing CRD in namespace: tekton-pipelines, version: v%s ", os.Getenv("PIPELINE_VERSION")))
+	cmdName := fmt.Sprintf("kubectl apply -f https://github.com/tektoncd/pipeline/releases/download/v%s/release.yaml", os.Getenv("PIPELINE_VERSION"))
```

Comment:
One problem with that is that it makes the e2e tests requiring a internet connection at all time, I wonder if we shouldn't add those yaml in a `testdata` folder under `test/e2e` :angel:

Reply:
This script is deprecated probably I have to remove it